arch/arc/arcmwdt/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_ARCMWDT_LIBC OR CONFIG_CPLUSPLUS)
+  zephyr_library_sources(arcmwdt-dtr-stubs.c)
```

Comment:
One more observation.
A `zephyr_library_sources()` without a prior `zephyr_library()` / `zephyr_library_named()` results in this file ending up in `libzephyr.a`.
Is that intended ?

If it is, then a `zephyr_sources(...)` is the right function to use here.

Reply:
Well, it wasn't done intentionally - we don't really care where exactly arcmwdt-dtr-stubs will be located so libzephyr.a is fully valid option. I've switch it to `zephyr_sources` usage.

Reply:
@tejlmand could you please check if you are OK with updated implementation?
Thanks!